pkg/apps/util/util.go

```diff
@@ -782,6 +788,23 @@ func DeploymentsForCleanup(configuration *deployapi.DeploymentConfig, deployment
 	return relevantDeployments
 }
 
+func ConfigExceededTimeoutSeconds(config *deployapi.DeploymentConfig, latestRC *v1.ReplicationController) bool {
+	var timeoutSeconds int64
+	if params := config.Spec.Strategy.RollingParams; params != nil {
+		timeoutSeconds = deployapi.DefaultRollingTimeoutSeconds
+		if params.TimeoutSeconds != nil {
+			timeoutSeconds = *params.TimeoutSeconds
+		}
+	}
+	if params := config.Spec.Strategy.RecreateParams; params != nil {
+		timeoutSeconds = deployapi.DefaultRecreateTimeoutSeconds
+		if params.TimeoutSeconds != nil {
+			timeoutSeconds = *params.TimeoutSeconds
+		}
+	}
+	return int64(time.Since(latestRC.CreationTimestamp.Time)*time.Second) > timeoutSeconds
```

Comment:
I need to check how we do rollbacks because in that case creationTimestamp might be an issue

Reply:
the RC has to be in "new" state for this to trigger... it is only in new when it is new :)

Reply:
Yeah, we do rollbacks differently from upstream anyway (we always create a new RC).